Original patch from gentoo: ../../dev/crosstool-NG/ct-ng.trunk/patches/strace/4.5.17/140-statfs64-check.patch
-= BEGIN original header =-
diff -dur strace-4.5.16.orig/acinclude.m4 strace-4.5.16/acinclude.m4
-= END original header =-
diff -durN strace-4.5.17.orig/acinclude.m4 strace-4.5.17/acinclude.m4
--- strace-4.5.17.orig/acinclude.m4	2004-04-14 04:21:01.000000000 +0200
+++ strace-4.5.17/acinclude.m4	2008-10-24 21:30:13.000000000 +0200
@@ -210,6 +210,26 @@
 fi
 ])
 
+dnl ### A macro to determine whether statfs64 is defined.
+AC_DEFUN([AC_STATFS64],
+[AC_MSG_CHECKING(for statfs64 in sys/(statfs|vfs).h)
+AC_CACHE_VAL(ac_cv_type_statfs64,
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#ifdef LINUX
+#include <linux/types.h>
+#include <sys/statfs.h>
+#else
+#include <sys/vfs.h>
+#endif]], [[struct statfs64 st;]])],[ac_cv_type_statfs64=yes],[ac_cv_type_statfs64=no])])
+AC_MSG_RESULT($ac_cv_type_statfs64)
+if test "$ac_cv_type_statfs64" = yes
+then
+	AC_DEFINE([HAVE_STATFS64], 1,
+[Define if statfs64 is available in sys/statfs.h or sys/vfs.h.])
+fi
+])
+
+
+
 dnl ### A macro to determine if off_t is a long long
 AC_DEFUN([AC_OFF_T_IS_LONG_LONG],
 [AC_MSG_CHECKING(for long long off_t)
diff -durN strace-4.5.17.orig/configure.ac strace-4.5.17/configure.ac
--- strace-4.5.17.orig/configure.ac	2008-07-22 02:42:29.000000000 +0200
+++ strace-4.5.17/configure.ac	2008-10-24 21:30:13.000000000 +0200
@@ -169,6 +169,7 @@
 		  struct stat.st_level,
 		  struct stat.st_rdev])
 AC_STAT64
+AC_STATFS64
 
 AC_TYPE_SIGNAL
 AC_TYPE_UID_T
diff -durN strace-4.5.17.orig/file.c strace-4.5.17/file.c
--- strace-4.5.17.orig/file.c	2008-05-20 06:56:18.000000000 +0200
+++ strace-4.5.17/file.c	2008-10-24 21:30:13.000000000 +0200
@@ -1703,7 +1703,7 @@
 	return 0;
 }
 
-#ifdef LINUX
+#ifdef HAVE_STATFS64
 static void
 printstatfs64(tcp, addr)
 struct tcb *tcp;
